File: 209-minimum-size-subarray-sum/minimum-size-subarray-sum.cpp
class Solution {
public:
    int minSubArrayLen(int target, vector<int>& nums) {
        int MinLenWindow = INT_MAX;
        int CurSum = 0;
        int curWindowSize = 0;
        // two pointer sliding window
        int low=0;
        int high=0;
        while(high<nums.size()){
            CurSum += nums[high];
            high++;
            while(CurSum >= target){
                curWindowSize = high - low;
                MinLenWindow = min(curWindowSize,MinLenWindow);
                CurSum -= nums[low];
                low++;
            }
        }
        if(MinLenWindow==INT_MAX){
            return 0;
        }
        return MinLenWindow;
    }
};

File: 209-minimum-size-subarray-sum/minimum-size-subarray-sum.java
class Solution {
    public int minSubArrayLen(int target, int[] nums) {
        int minLength = Integer.MAX_VALUE;
        int start = 0;
        int end = 0;
        int sum = 0;
        while(end< nums.length){
            sum+= nums[end];
            end++;
            while(sum >= target){
                int len = end-start;
                minLength = Math.min(minLength,len);
                sum -= nums[start];
                start++;
            }
        }
        if(minLength == Integer.MAX_VALUE){
            return 0;
        }
        return minLength;
    }
}